src/TaskLineRenderer.ts

```diff
@@ -217,11 +240,14 @@ async function taskToHtml(
                 addInternalClasses(component, internalSpan);
 
                 // Add the component's CSS class describing what this component is (priority, due date etc.)
-                const componentClass = getTaskComponentClass(component);
-                span.classList.add(...componentClass);
+                const fieldLayoutDetails = FieldLayouts[component];
+                if (fieldLayoutDetails) {
+                    const componentClass = [fieldLayoutDetails.className];
+                    span.classList.add(...componentClass);
+                }
 
                 // Add the component's attribute ('priority-medium', 'due-past-1d' etc.)
-                const componentDataAttribute = getComponentDataAttribute(component, task);
+                const componentDataAttribute = FieldLayouts[component].getDataAttribute(component, task);
```

Comment:
Yup, I have it already done but didn't include in this PR. The result is:

```ts
const componentDataAttribute = getComponentDataAttribute(component, task);
```

I suggest merging as is and having that in a separate PR.

Reply:
Sounds good. Let me know when you are done with this and I will merge...

Reply:
Nope, sorry, I'm wrong, this is not changed later. I agree that it would be nice to remove.

This is an issue indeed. The problem is that even if FLDs are declared as:

```ts
export const FieldLayouts: { [c in TaskLayoutComponent]: FieldLayoutDetail } = {
    createdDate: new FieldLayoutDetail('task-created', 'taskCreated', dateDataAttributeCalculator),
    dueDate: new FieldLayoutDetail('task-due', 'taskDue', dateDataAttributeCalculator),
...
};
```

We don't have access to an object's name, in other words, for `const example = FieldLayouts['createdDate']` object, there is no way to get the `createdDate` value.

I thought about adding component name parameter to the `FieldLayoutDetail` class but this would lead to initialisers like:

export const FieldLayouts: { [c in TaskLayoutComponent]: FieldLayoutDetail } = {
    createdDate: new FieldLayoutDetail(`createdDate`, 'task-created', 'taskCreated', dateDataAttributeCalculator),
    dueDate: new FieldLayoutDetail('dueDate', 'task-due', 'taskDue', dateDataAttributeCalculator),
...
};
```

which is ugly and kills the use of the dictionary. Please let me know what you think could be a good solution here.

Reply:
The ideal thing would be something like

```ts
const componentDataAttribute = FieldLayouts(get_component_somehow).getDataAttribute(task);
```

Reply:
I think part of the problem here is that a raw list/array/whatever is being exported - `FieldLayouts` - rather than being encapsulated in a class....

By exporting a list of values, all users of it have to understand about the storage and interact with the stored.

If `FieldLayouts` data was in a class, for example, and its constructor initialised the values, then it can have nice simple methods that take a Task and a component, and any mildly-smelly duplicate references to the component is hidden inside the method implementations.

Reply:
I had roughly the same thing in mind. I will try to make an implementation in this PR. Meanwhile converting this to a draft.

Does something like this looks ok? This is a draft obviously.

```ts
class FieldLayoutsContainer {
    const container = new AttributesDictionary();

   constructor(attributesDictionary: AttributesDictionary) {
        this.container = attributesDictionary;
    }

    public getDetails(component: TaskLayoutComponent): FieldLayoutDetails {
        ...
    }
}

const fieldLayoutsContainer = new FieldLayoutsContainer(FieldLayouts);

...

const fieldLayoutDetail = fieldLayoutsContainer.getDetails(component);
const dataAttribute = fieldLayoutDetail.getDataAttribute(task);

...
```

Reply:
Yes, that's the kind of thing... I wouldn't put `Container` on the class name - and I wouldn't put `get` on the method names... But basically, yes...

However, I think if/as you're happy with this principle, it's better to merge the current code, and then add the class as a separate PR...

Then the reviewing and conversation we have had is in the bag, so to speak. And the next PR is smaller and easier to review...

Reply:
But not this:

```ts
   constructor(dictionary: /* TaskLayoutComponent : FieldLayoutDetail dictionary */) {
        this.container = dictionary;
    }
```

The point is to encapsulate the data... so the constructor takes nothing, and it initialises itself.

Reply:
> However, I think if/as you're happy with this principle, it's better to merge the current code, and then add the class as a separate PR...

Yes, sure. Then what would be needed to be done in this PR to be merged?

Reply:
Thank you. 

1. If rebasing gives no conflicts, and you have time, please could you rebase. 
2. Then push
3. Then remove the Draft status on this

Reply:
Rebased (no conflicts) and force pushed =)